### ISSUE TITLE
InvalidParameterValueException: Unzipped size must be smaller than 262144000 bytes 

### ISSUE THREAD

**Issue by pr1ntr** [artilleryio/artillery#2419]
Version info: 
```
Artillery: 2.0.4
Node.js:   v18.18.0
OS:        win32
```

Running this command: 
```
yarn artillery run-lambda --region us-west-2 xxxxx.dev.yml --output dev.server.json
```

I expected to see this happen:
It deploys to lambda

Instead, this happened:
```
InvalidParameterValueException: Unzipped size must be smaller than 262144000 bytes
    at Object.extractError (C:\Users\xxxx\AppData\Local\Yarn\Data\global\node_modules\aws-sdk\lib\protocol\json.js:80:27)
    at Request.extractError (C:\Users\xxxx\AppData\Local\Yarn\Data\global\node_modules\aws-sdk\lib\protocol\rest_json.js:61:8)
    at Request.callListeners (C:\Users\xxxx\AppData\Local\Yarn\Data\global\node_modules\aws-sdk\lib\sequential_executor.js:106:20)
    at Request.emit (C:\Users\xxxx\AppData\Local\Yarn\Data\global\node_modules\aws-sdk\lib\sequential_executor.js:78:10)
    at Request.emit (C:\Users\xxxx\AppData\Local\Yarn\Data\global\node_modules\aws-sdk\lib\request.js:686:14)
    at Request.transition (C:\Users\xxxx\AppData\Local\Yarn\Data\global\node_modules\aws-sdk\lib\request.js:22:10)
    at AcceptorStateMachine.runTo (C:\Users\xxxx\AppData\Local\Yarn\Data\global\node_modules\aws-sdk\lib\state_machine.js:14:12)
    at C:\Users\xxxx\AppData\Local\Yarn\Data\global\node_modules\aws-sdk\lib\state_machine.js:26:10
    at Request.<anonymous> (C:\Users\xxxx\AppData\Local\Yarn\Data\global\node_modules\aws-sdk\lib\request.js:38:9)
    at Request.<anonymous> (C:\Users\xxxx\AppData\Local\Yarn\Data\global\node_modules\aws-sdk\lib\request.js:688:12) {
  code: 'InvalidParameterValueException',
  '[Type]': 'See error.Type for details.',
  time: 2024-01-16T18:20:15.976Z,
  requestId: '48629e5f-9f5c-4fe4-bf9d-222c945e5747',
  statusCode: 400,
  retryable: false,
  retryDelay: 24.66141534760702
}
```

Files being used:
```
config:
  target: "https://xxxxxxxxxxxxxxx.com"
  phases:
    - duration: 10m
      arrivalRate: 250
      rampTo: 0
      name: 'hammer'


scenarios:
  - flow:
      - loop:
          - options:
              url: "/submit"
              headers:
                "User-Agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/120.0.0.0 Safari/537.36"
                "Access-Control-Request-Method": "post"
                "Origin": "https://xxxxxxxxxxxxxxxx.com"
          - post:
              url: "/submit"
              json:
                fullName: "{{$randomString(7)}} {{$randomString(7)}}"
                email: "{{$randomString(7)}}@{{$randomString(7)}}.com"
                code: "{{$randomString(22)}}"
              headers:
                "Content-Type": "application/json"
                "User-Agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/120.0.0.0 Safari/537.36"
                "Origin": "xxxxxxxxxxxxxxxxx.com"

        count: 100
```

package.json
```
{
  "name": "xxxxxx",
  "version": "1.0.0",
  "main": "index.js",
  "author": "@wildlifela",
  "license": "NO LICENSE",
  "private": true,
  "scripts": {
    "run:dev": "artillery run xxxxx.dev.yml",
    "run:dev:lambda": "artillery run-lambda --region us-west-2 xxxxxx.dev.yml --output dev.server.json"
  }
}
```
[package-lock.json](https://github.com/artilleryio/artillery/files/13954463/package-lock.json)

I am not sure what is going on with this, I have no dependencies in my package.json. the unzipped package on s3 is 518mb. I uploaded the package-lock from the lambda. It seems to including every package imaginable so no wonder the contents are so large. I don't know how to proceed here. Any help is appreciated.

**Comment by bernardobridge** [comment]
Hi @pr1ntr 👋 ,

Thanks for reporting this. We are aware of this issue in certain cases, and right now are exploring options to improve the Lambda setup. It's interesting it's happening to you without having any dependencies, I don't think I've seen that before 🤔 

In the meantime, you can try to use Fargate support, which is more robust and should fit most use cases. I've given you some pointers in the other issue you opened. Thank you!

**Comment by pr1ntr** [comment]
It is strange, in the past I have created robust lambdas using Docker containers. That was pretty straight forward with CDK.

**Comment by bernardobridge** [comment]
👋 This should be fixed in [Artillery v2.0.5](https://github.com/artilleryio/artillery/releases/tag/artillery-2.0.5). Please let us know if you still run into issues after.

However `artillery-plugin-publish-metrics` will no longer be compatible with Lambda for the time being.
